packages/sanity/src/_internal/manifest/extractManifest.ts

```diff
@@ -0,0 +1,499 @@
+import startCase from 'lodash/startCase'
+import {
+  type ArraySchemaType,
+  type BlockDefinition,
+  type BooleanSchemaType,
+  ConcreteRuleClass,
+  createSchema,
+  type CrossDatasetReferenceSchemaType,
+  type FileSchemaType,
+  type MultiFieldSet,
+  type NumberSchemaType,
+  type ObjectField,
+  type ObjectSchemaType,
+  type ReferenceSchemaType,
+  type Rule,
+  type RuleSpec,
+  type Schema,
+  type SchemaType,
+  type SchemaValidationValue,
+  type SpanSchemaType,
+  type StringSchemaType,
+  type Workspace,
+} from 'sanity'
+
+import {
+  getCustomFields,
+  isCrossDatasetReference,
+  isCustomized,
+  isDefined,
+  isPrimitive,
+  isRecord,
+  isReference,
+  isString,
+  isType,
+} from './manifestTypeHelpers'
+import {
+  type ManifestField,
+  type ManifestFieldset,
+  type ManifestSchemaType,
+  type ManifestSerializable,
+  type ManifestTitledValue,
+  type ManifestValidationGroup,
+  type ManifestValidationRule,
+  type ManifestWorkspace,
+} from './manifestTypes'
+
+interface Context {
+  schema: Schema
+}
+
+type SchemaTypeKey =
+  | keyof ArraySchemaType
+  | keyof BooleanSchemaType
+  | keyof FileSchemaType
+  | keyof NumberSchemaType
+  | keyof ObjectSchemaType
+  | keyof StringSchemaType
+  | keyof ReferenceSchemaType
+  | keyof BlockDefinition
+  | 'group' // we strip this from fields
+
+type Validation = {validation: ManifestValidationGroup[]} | Record<string, never>
+type ObjectFields = {fields: ManifestField[]} | Record<string, never>
+type SerializableProp = ManifestSerializable | ManifestSerializable[] | undefined
+type ManifestValidationFlag = ManifestValidationRule['flag']
+type ValidationRuleTransformer = (rule: RuleSpec) => ManifestValidationRule | undefined
+
+const MAX_CUSTOM_PROPERTY_DEPTH = 5
+const INLINE_TYPES = ['document', 'object', 'image', 'file']
+
+export function extractWorkspace(workspace: Workspace): ManifestWorkspace {
+  const serializedSchema = extractManifestSchemaTypes(workspace.schema)
+
+  return {
+    name: workspace.name,
+    dataset: workspace.dataset,
+    schema: serializedSchema,
+  }
+}
```

Comment:
Would it be possible to expose workspace `title` and `subtitle` too? (Which may be more meaningful for users when selecting workspaces in _Create_)

Reply:
Another late addition to the above – I believe we'll also need to include workspace `basePath` (in order to construct preview URLs in _Create_)

Re: custom providers – I don't see the harm in including, provided it's low effort to to so!

Reply:
We will have title, subtitle and basePath on each workspace. 
We will not have auth providers at this point, as the studio supports functions for these.